### ISSUE TITLE
Excessive log messages

### ISSUE THREAD

**Issue by jhelwig** [Celeo/rfesi#36]
Having `rfesi` log errors results in a lot of unnecessary messages, when I'm handling them in the code that's calling `rfesi`.

For example: `rfesi` logs every `404`, but intentionally encountering `404`s is the only way to make sure that every page of data is retrieved until the response headers (#34) are exposed to end users.

<details>
<summary>Wrapper to retrieve every page</summary>

```rust
    #[derive(Debug)]
    pub enum EveClientError {
        Esi(EsiError),
        Other(anyhow::Error),
    }

    impl From<EsiError> for EveClientError {
        fn from(value: EsiError) -> Self {
            Self::Esi(value)
        }
    }

    impl From<anyhow::Error> for EveClientError {
        fn from(value: anyhow::Error) -> Self {
            Self::Other(value)
        }
    }

    impl From<EveClientError> for anyhow::Error {
        fn from(value: EveClientError) -> Self {
            match value {
                EveClientError::Esi(esi_error) => anyhow!(esi_error),
                EveClientError::Other(other) => other,
            }
        }
    }

    pub async fn fetch_all_pages<T, L>(mut fetch_page: L) -> Result<Vec<T>>
    where
        L: AsyncFnMut<(i32,), Output = std::result::Result<Vec<T>,EveClientError>>
    {
        let mut results = Vec::<T>::new();
        let mut current_page = 1;

        loop {
            match fetch_page.call_mut((current_page,)).await {
                Ok(page_results) => results.extend(page_results),
                Err(EveClientError::Esi(EsiError::InvalidStatusCode(404))) => break,
                Err(e) => return Err(e.into()),
            }
            current_page += 1;
        }

        Ok(results)
    }
```
</details>

<details>
<summary>Example usage</summary>

```rust
        let mut esi = EsiBuilder::new()
            .user_agent("Your user agent")
            .build()?;
        esi.update_spec().await?;

        let new_market_data = crate::eve_client::fetch_all_pages(
            async_owned_closure_mut!(
                {
                    esi: Esi = esi,
                    region: i32 = region,
                };
                async |page_no: i32| -> std::result::Result<Vec<MarketOrder>, EveClientError> {
                    debug!("Fetching market orders for region {region} page {page_no}.");
                    esi.group_market().get_region_orders(*region, None, Some(page_no), None).await.map_err(Into::into)
                }
            )
        ).await?;
```
</details>

It seems that having `rfesi` return a `Result` (as it already does) should be sufficient, and it shouldn't need to also log at the error level internally, especially when it's an error that is able to be handled by the user of the library.

**Comment by Celeo** [comment]
Yup, fair.

**Comment by Celeo** [comment]
I've removed the log statement for normal queries, and reduced the level of a few other log statements: 06b270f85fb535f6cfea3a1c7161f68fb89e508b. Released in [0.39.1](https://github.com/Celeo/rfesi/releases/tag/v0.39%2C1).

Let me know if this works better for you.

As a sidenote, I'm learning more just by looking at those snippets you shared, so thanks for that!

**Comment by Celeo** [comment]
Let me know if there's anything missing from this effort, and I'll happily address. Thanks!
